### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
MSVC:
* reg.  declaration of 'dummy' hides global declaration (C4459)

GCC:
* reg. this statement may fall through [-Wimplicit-fallthrough=]
* reg. comparison of integer expressions of different signedness: ‘int32_t’ and 'long unsigned int’ [-Wsign-compare]
* reg. passing argument 2 to ‘restrict’-qualified parameter aliases with argument 1 [-Wrestrict]
* reg. unrecognized command-line option ‘-Wno-tautological-pointer-compare’ may have been intended to silence earlier diagnostics
* reg. command-line option ‘-Wno-absolute-value’ is valid for C/ObjC but not for C++
* reg. deprecated [-Wdeprecated-declarations]

### CODE REVIEW

**Comment by lyrra** [comment]
Time for -Werror ? 

If you're hunting warnings, GCC sees these:

```
[  1%] Building C object src/framework/audio/opusenc/opus/CMakeFiles/opus.dir/src/opus_decoder.c.o
/d/shrek/ms/ms4/thirdparty/opus/src/opus_decoder.c:37:10: note: ‘#pragma message: You appear to be compiling without optimization, if so opus will be very slow.’
   37 | # pragma message "You appear to be compiling without optimization, if so opus will be very slow."
      |          ^~~~~~~
```

```
[  8%] Building C object src/framework/audio/opusenc/opus/CMakeFiles/opus.dir/silk/NLSF_VQ.c.o
/d/shrek/ms/ms4/thirdparty/google_crashpad_client/client/crashpad_client_linux.cc: In lambda function:
/d/shrek/ms/ms4/thirdparty/google_crashpad_client/client/crashpad_client_linux.cc:460:31: warning: passing argument 2 to ‘restrict’-qualified parameter aliases with argument 1 [-Wrestrict]
  460 |       if (sigaltstack(&stack, &stack) != 0) {
      |                       ~~~~~~  ^~~~~~
```

```
[ 32%] Building C object src/framework/draw/freetype/CMakeFiles/freetype.dir/freetype-2.13.1/src/base/ftwinfnt.c.o
/d/shrek/ms/ms4/thirdparty/fluidsynth/fluidsynth-2.3.3/src/midi/fluid_midi.c: In function ‘fluid_player_set_bpm’:
/d/shrek/ms/ms4/thirdparty/fluidsynth/fluidsynth-2.3.3/src/midi/fluid_midi.c:2542:5: warning: ‘fluid_player_set_midi_tempo’ is deprecated [-Wdeprecated-declarations]
 2542 |     return fluid_player_set_midi_tempo(player, 60000000L / bpm);
      |     ^~~~~~
/d/shrek/ms/ms4/thirdparty/fluidsynth/fluidsynth-2.3.3/src/midi/fluid_midi.c:2519:5: note: declared here
 2519 | int fluid_player_set_midi_tempo(fluid_player_t *player, int tempo)
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~

```

```
[ 42%] Building CXX object src/framework/audio/lame/CMakeFiles/lame.dir/lame_autogen/mocs_compilation.cpp.o
cc1plus: warning: command-line option ‘-Wno-absolute-value’ is valid for C/ObjC but not for C++
[ 42%] Building C object src/framework/audio/flac/CMakeFiles/flac.dir/flac-1.4.3/src/libFLAC/fixed.c.o
[ 42%] Building C object src/framework/audio/flac/CMakeFiles/flac.dir/flac-1.4.3/src/libFLAC/fixed_intrin_avx2.c.o
[ 42%] Building C object src/framework/audio/flac/CMakeFiles/flac.dir/flac-1.4.3/src/libFLAC/fixed_intrin_sse2.c.o
cc1plus: note: unrecognized command-line option ‘-Wno-tautological-pointer-compare’ may have been intended to silence earlier diagnostics
```

```
[ 50%] Building CXX object src/framework/audio/CMakeFiles/audio.dir/audio_autogen/mocs_compilation.cpp.o
cc1plus: warning: /d/shrek/ms/ms4/builds/Linux-Qt-usr-Make-Debug/src/framework/global/CMakeFiles/global.dir/cmake_pch.hxx.gch: not used because `STDC_HEADERS' is defined [-Winvalid-pch]
[ 51%] Linking CXX static library libmultiinstances.a
[ 51%] Building CXX object src/framework/audio/CMakeFiles/audio.dir/Unity/unity_4_cxx.cxx.o
cc1plus: warning: /d/shrek/ms/ms4/builds/Linux-Qt-usr-Make-Debug/src/framework/global/CMakeFiles/global.dir/cmake_pch.hxx.gch: not used because `STDC_HEADERS' is defined [-Winvalid-pch]
[ 51%] Building CXX object src/framework/audio/CMakeFiles/audio.dir/Unity/unity_3_cxx.cxx.o
cc1plus: warning: /d/shrek/ms/ms4/builds/Linux-Qt-usr-Make-Debug/src/framework/global/CMakeFiles/global.dir/cmake_pch.hxx.gch: not used because `STDC_HEADERS' is defined [-Winvalid-pch]
[ 51%] Building CXX object src/framework/audio/CMakeFiles/audio.dir/Unity/unity_2_cxx.cxx.o
cc1plus: warning: /d/shrek/ms/ms4/builds/Linux-Qt-usr-Make-Debug/src/framework/global/CMakeFiles/global.dir/cmake_pch.hxx.gch: not used because `STDC_HEADERS' is defined [-Winvalid-pch]
[ 51%] Building CXX object src/framework/audio/CMakeFiles/audio.dir/Unity/unity_1_cxx.cxx.o
cc1plus: warning: /d/shrek/ms/ms4/builds/Linux-Qt-usr-Make-Debug/src/framework/global/CMakeFiles/global.dir/cmake_pch.hxx.gch: not used because `STDC_HEADERS' is defined [-Winvalid-pch]
[ 51%] Building CXX object src/framework/audio/CMakeFiles/audio.dir/Unity/unity_0_cxx.cxx.o
cc1plus: warning: /d/shrek/ms/ms4/builds/Linux-Qt-usr-Make-Debug/src/framework/global/CMakeFiles/global.dir/cmake_pch.hxx.gch: not used because `STDC_HEADERS' is defined [-Winvalid-pch]
[ 51%] Linking CXX static library libdraw.a
```

```
[ 57%] Linking CXX executable accessibility_tests
/d/shrek/ms/ms4/src/engraving/libmscore/chord.cpp: In member function ‘virtual void mu::engraving::Chord::remove(mu::engraving::EngravingItem*)’:
/d/shrek/ms/ms4/src/engraving/libmscore/chord.cpp:807:9: warning: this statement may fall through [-Wimplicit-fallthrough=]
  807 |         }
      |         ^
/d/shrek/ms/ms4/src/engraving/libmscore/chord.cpp:810:5: note: here
  810 |     case ElementType::CHORDLINE:
      |     ^~~~
```

```
[ 80%] Automatic MOC and UIC for target palette
In file included from /d/shrek/ms/ms4/builds/Linux-Qt-usr-Make-Debug/src/importexport/guitarpro/CMakeFiles/iex_guitarpro.dir/Unity/unity_1_cxx.cxx:7:
/d/shrek/ms/ms4/src/importexport/guitarpro/internal/gtp/gpconverter.cpp: In member function ‘void mu::iex::guitarpro::GPConverter::addBarline(const mu::iex::guitarpro::GPMasterBar*, mu::iex::guitarpro::GPConverter::Measure*, int32_t)’:
/d/shrek/ms/ms4/src/importexport/guitarpro/internal/gtp/gpconverter.cpp:446:40: warning: comparison of integer expressions of different signedness: ‘int32_t’ {aka ‘int’} and ‘std::vector<std::unique_ptr<mu::iex::guitarpro::GPMasterBar> >::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
  446 |     bool lastMeasure = (masterBarIndex == _gpDom->masterBars().size() - 1);
      |                         ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by Jojo-Schmitz** [comment]
Tried to fix some of those gcc warning, please check.
And be my guess for fixing more ;-)

File: build/cmake/SetupCompileWarnings.cmake [hunk 1]
```diff
@@ -50,6 +50,8 @@ function(target_no_warning TARGET WNAME)
     elseif(WNAME STREQUAL "-Wno-tautological-pointer-compare")
     elseif(WNAME STREQUAL "-Wno-sign-compare")
     elseif(WNAME STREQUAL "-Wno-attributes")
+    elseif(WNAME STREQUAL "-Wno-restrict")
+    elseif(WNAME STREQUAL "-Wno-deprecated-declaraytions")
```
Comment (lyrra) [inline build/cmake/SetupCompileWarnings.cmake:54]: typo

Comment (Jojo-Schmitz) [reply build/cmake/SetupCompileWarnings.cmake:54]: yup, just fiound it myself

**Comment by igorkorsukov** [comment]
> reg. declaration of 'dummy' hides global declaration (C4459)

I believe that we need to find this global variable `dummy` and at best make it non-global, or at least rename it to something like `g_dummy`, because there can be many local variables `dummy`, they are quite often used.

**Comment by Jojo-Schmitz** [comment]
There's lots of them, many appear generated (filenames "qrc_*.cpp"), all of the form
```cpp
namespace {
   struct initializer {
       initializer() { QT_RCC_MANGLE_NAMESPACE(qInitResources_fonts_FinaleBroadway)(); }
       ~initializer() { QT_RCC_MANGLE_NAMESPACE(qCleanupResources_fonts_FinaleBroadway)(); }
   } g_dummy;
```

**Comment by Jojo-Schmitz** [comment]
And this actually is the one if collides with there

**Comment by igorkorsukov** [comment]
> There's lots of them, all appear generated files (name "qrc_*.cpp"), all of the form
> 
> ```c++
> namespace {
>    struct initializer {
>        initializer() { QT_RCC_MANGLE_NAMESPACE(qInitResources_fonts_FinaleBroadway)(); }
>        ~initializer() { QT_RCC_MANGLE_NAMESPACE(qCleanupResources_fonts_FinaleBroadway)(); }
>    } dummy;
> ```

oh (

**Pull request merged**